experimental/HttpBenchmark/Program.cs

```diff
@@ -0,0 +1,380 @@
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        private static void Init(int connections, TimeSpan duration, int pipeline, bool keepalive, Uri uri)
+        {
+#if DEBUG
+            Console.WriteLine($"Configuration: Debug");
+#else
+            Console.WriteLine($"Configuration: Release");
+#endif
+
+            var gc = GCSettings.IsServerGC ? "server" : "client";
+            Console.WriteLine($"GC: {gc}");
+
+            Console.WriteLine($"Connections: {connections}");
+            Console.WriteLine($"Duration: {duration}");
+            Console.WriteLine($"Pipeline: {pipeline}");
+            Console.WriteLine($"Keepalive: {keepalive}");
+            Console.WriteLine($"Uri: {uri}");
+
+            Console.WriteLine();
+        }
+
+        private static async Task<Socket> CreateSocket(Uri uri)
+        {
+            var tcs = new TaskCompletionSource<Socket>();
+
+            var socketArgs = new SocketAsyncEventArgs();
+            socketArgs.RemoteEndPoint = new DnsEndPoint(uri.DnsSafeHost, uri.Port);
+            socketArgs.Completed += (s, e) => tcs.TrySetResult(e.ConnectSocket);
+
+            // Must use static ConnectAsync(), since instance Connect() does not support DNS names on OSX/Linux.
+            if (Socket.ConnectAsync(SocketType.Stream, ProtocolType.Tcp, socketArgs))
+            {
+                await tcs.Task;
+            }
+
+            var socket = socketArgs.ConnectSocket;
+
+            if (socket == null)
+            {
+                throw new SocketException((int)socketArgs.SocketError);
+            }
+            else
+            {
+                return socket;
+            }
+        }
+
+        private static void SendReceive(Socket socket, byte[] requestBytes, int responseLength, byte[] responseBuffer)
+        {
+            socket.Send(requestBytes);
+
+            int bytesReceived = 0;
+            while (bytesReceived < responseLength)
+            {
+                bytesReceived += socket.Receive(responseBuffer, bytesReceived, responseBuffer.Length - bytesReceived, SocketFlags.None);
+            }
+        }
+
+        private static async Task WriteResults(CancellationToken cancellationToken)
+        {
+            var lastRequests = (long)0;
+            var lastElapsed = TimeSpan.Zero;
+
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                await Task.Delay(TimeSpan.FromSeconds(1));
+
+                var currentRequests = _requests - lastRequests;
+                lastRequests = _requests;
+
+                var elapsed = _stopwatch.Elapsed;
+                var currentElapsed = elapsed - lastElapsed;
+                lastElapsed = elapsed;
+
+                WriteResult(_requests, currentRequests, currentElapsed);
+            }
+
+            Console.WriteLine();
+            Console.WriteLine($"Average RPS: {Math.Round(_requests / _stopwatch.Elapsed.TotalSeconds)}");
```

Comment:
nit: this not a complain but more like a feature request. can we save the result in a json file. it would be useful for pipeline the tools.